---
title: "Assignment_week3"
author: "Beata Vekeriotaite"
date: "3/14/2022"
output: html_document
---

# So, the code is here but something keeps glitching and so I can't get to the end of it. Hence, uploading the code only

```{r, eval=FALSE}
suppressPackageStartupMessages({
  library(AnnotationHub)
  library(Rsubread)
  library(rtracklayer)
  library(Biostrings)
  library(Rfastp)
  library(epiwraps)
  library(GenomicRanges)
})

ah <- AnnotationHub()

setwd("/home/bvekeriotait/public/BeataVekeriotaite/Bioinformatics course 2022/Practice/Assignments/materials2022")

```

# Download a raw mouse ChIPseq dataset (p300 in mESC)

```{r, eval=FALSE}
dir.create("raw")
download.file("https://www.encodeproject.org/files/ENCFF001KEU/@@download/ENCFF001KEU.fastq.gz", 
              dest="raw/p300.fastq.gz")
```

# Process the raw data

## Reads QC and trimming

```{r, eval=FALSE}
dir.create("rfastp.trimmed")
qc <- Rfastp::rfastp("raw/p300.fastq.gz", outputFastq="rfastp.trimmed/p300", 
                    thread=4, overrepresentationAnalysis=TRUE)
```

```{r}
Rfastp::curvePlot(qc, curve="content_curves")
```

## Building a genome index for mapping --> done on server, was taking forever otherwise

```{r, eval=FALSE}

# downloading the mouse genome sequence (newest one) from AnnotationHub
mouse_genome <- ah[["AH14006"]]

# creating a new directory containing Mus Musculus genome
dir.create("Mus_musculus_genome")

# writing a genome sequence in fasta format
export(import.2bit(mouse_genome), "Mus_musculus_genome/mouse_genome.fasta.gz", compress=TRUE)

# building an Rsubread index
Rsubread::buildindex("Mus_musculus_genome/rsubread", reference="Mus_musculus_genome/mouse_genome.fasta.gz")

```

## Alignment - bam file

```{r}
dir.create("aligned")
align.stats <- Rsubread::align(index="Mus_musculus_genome/rsubread", type="dna",
                               output_file="aligned/p300.bam",
                               readfile1="rfastp.trimmed/p300_R1.fastq.gz", 
                               nthreads=4, sortReadsByCoordinates=TRUE)
align.stats
```



# Obtain peaks

```{r}

# it was taking forever to get the peaks so I ran it on the server as following:
# 
# setwd("\\hest.nas.ethz.ch\green_groups_NME_public\BeataVekeriotaite\Bioinformatics course 2022\Practice\Assignments\materials2022")
# 
# library(epiwraps)
# 
# peaks <- callPeaks("aligned/p300.bam")
# 
# saveRDS(peaks, "calledMACSPeaks.rds")


#peaks <- callPeaks("aligned/p300.bam")


peaks <- readRDS("~/public/BeataVekeriotaite/Bioinformatics course 2022/Practice/Assignments/materials2022/calledMACSPeaks.rds")

```

# Generating coverage tracks - bigwig file

```{r}
# just as above, I ran this on the server as following:
# 
# library(epiwraps)
# 
# dir.create("tracks")
# 
# bam2bw("aligned/p300.bam", "tracks/p300.bw", binWidth=10, extend=100, scaling=TRUE)

```


# How many peaks do you find?

```{r}

length(peaks)

# strong_peaks <- peaks[peaks$p.value>median(peaks$p.value)]
# 
# peakSignals <- signal2Matrix("tracks/p300.bw", regions=strong_peaks, extend=2000, w=20)
# 
# plotEnrichedHeatmaps(peakSignals)
# 
# length(strong_peaks)

```


# Plot the signal around one of the peaks


```{r}

plotSignalTracks(c(signal="tracks/p300.bw", peaks=peaks), region = peaks[1])

```

# CHIP data 

```{r}

# bed-like format ENCODE data for H3K4me3 in mESCs

H3K4me3 <- import("/home/bvekeriotait/public/BeataVekeriotaite/Bioinformatics course 2022/Practice/Assignments/materials2022/chip/ENCFF619XDG_H3K4me3_ ES-Bruce4.bigBed")


# bed-like format ENCODE data for H3K4me1 in mESCs

H3K4me1 <- import("/home/bvekeriotait/public/BeataVekeriotaite/Bioinformatics course 2022/Practice/Assignments/materials2022/chip/ENCFF187LHP_H3K4me1_ES-Bruce4.bigBed")


# bed-like format ENCODE data for H3K27ac in mESCs

H3K27ac <- import("/home/bvekeriotait/public/BeataVekeriotaite/Bioinformatics course 2022/Practice/Assignments/materials2022/chip/ENCFF583TQN_H3K27ac_ES-Bruce4.bigBed")


# bed-like format ENCODE data for H3K27me3 in mESCs

H2K27me3 <- import("/home/bvekeriotait/public/BeataVekeriotaite/Bioinformatics course 2022/Practice/Assignments/materials2022/chip/ENCFF472QPM_H3K27me3_ ES-Bruce4.bigBed")

```

# Of the p300 peaks you identified last week, what proportion overlap each of the marks?


```{r}

# H3K4me3

p300_H3K4me3_overlaps <- overlapsAny(peaks, H3K4me3)

head(p300_H3K4me3_overlaps)

length(p300_H3K4me3_overlaps)


# H3K4me1

p300_H3K4me1_overlaps <- overlapsAny(peaks, H3K4me1)

head(p300_H3K4me1_overlaps)

length(p300_H3K4me1_overlaps)


# H3K27ac

p300_H3K27ac_overlaps <- overlapsAny(peaks, H3K27ac)

head(p300_H3K27ac_overlaps)

length(p300_H3K27ac_overlaps)


# H2K27me3

p300_H2K27me3_overlaps <- overlapsAny(peaks, H2K27me3)

head(p300_H2K27me3_overlaps)

length(p300_H2K27me3_overlaps)


```